pkg/providers/pricing/suite_test.go

```diff
@@ -262,6 +262,57 @@ var _ = Describe("Pricing", func() {
 		Expect(lo.Map(inp.ProductDescriptions, func(x *string, _ int) string { return *x })).
 			To(ContainElements("Linux/UNIX", "Linux/UNIX (Amazon VPC)"))
 	})
+	It("should return static on-demand data when in isolated-vpc", func() {
+		ctx = options.ToContext(ctx, test.Options(test.OptionsFields{
+			IsolatedVPC: lo.ToPtr(true),
+		}))
+		awsEnv.PricingAPI.GetProductsOutput.Set(&awspricing.GetProductsOutput{
+			PriceList: []aws.JSONValue{
+				fake.NewOnDemandPrice("c98.large", 1.20),
+				fake.NewOnDemandPrice("c99.large", 1.23),
+			},
+		})
+		// this should fail as we are not providing spot pricing info which will be
+		// called even when isolated-vpc is set
+		ExpectReconcileFailed(ctx, controller, types.NamespacedName{})
```

Comment:
Should we just check the `pricingProvider.SpotPrice` error that is returned? so that we can validate the exact error.